src/sbvr-api/sbvr-utils.ts

```diff
@@ -1205,11 +1299,20 @@ const runODataRequest = (req: Express.Request, vocabulary: string) => {
 			await runHooks('PREPARSE', reqHooks, { req, tx: req.tx });
 			let requests: uriParser.UnparsedRequest[];
 			// Check if it is a single request or a batch
-			if (req.batch != null && req.batch.length > 0) {
-				requests = req.batch;
+			if (req.url.startsWith(`/${vocabulary}/$batch`)) {
+				await Promise.all(
+					req.body.requests.map(
+						async (request: HookReq) =>
+							await runHooks('PREPARSE', reqHooks, {
+								req: request,
+								tx: req.tx,
+							}),
+					),
+				);
+				requests = req.body.requests;
```

Comment:
This will execute all requests in one database transaction. Is this desired or should every request run in it's own database transaction?
The background is, if a database transaction fails the qhole transaction is rolled back, thus all batch requests will be rolled back.

@Page- What do you think, should every request in a batch run in its own db tx?

Reply:
Yes, that is part of the point/idea of implementing $batch @fisehara , to run all requests in a batch on one transaction. Part of the following roadmap item: https://roadmap.balena.io/posts/19/allow-changes-to-variables-to-be-batched-and-submitted-all-at-once